### PR TITLE
[Feature] Security flag to force encryption at rest

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

Add a new security flag which allows forcing encryption at rest to be enabled. If the flag is enabled, then encryption at rest is enabled when the user session is created. Additionally, the option in the account settings is hidden to prevent the feature being disabled.

### Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-build-configuration/pull/43

### CODE REVIEW

File: Wire-iOS/Sources/AppRootViewController.swift [hunk 1]
```diff
@@ -556,6 +556,10 @@ extension AppRootViewController: SessionManagerCreatedSessionObserver, SessionMa
                 soundEventListeners[accountId] = SoundEventListener(userSession: userSession)
             }
         }
+
+        if SecurityFlags.forceEncryptionAtRest.isEnabled && !userSession.encryptMessagesAtRest {
+            userSession.encryptMessagesAtRest = true
+        }
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/AppRootViewController.swift:562]: @typfel I'm wondering if this is the best place to do this: when we will migrate the existing database to and from EAR, we might need to display some progress indication UI. I'm wondering if it's better to do this in the `SessionManager` where we have more control. What do you think?

Comment (billypchan) [reply Wire-iOS/Sources/AppRootViewController.swift:562]: create a method like `userSession.setEncryptMessagesAtRestIfNeeded()` would be better.

File: Wire-iOS/Wire-Info.plist [hunk 2]
```diff
@@ -206,5 +206,7 @@
 	<string>${CAMERA_ROLL_ENABLED}</string>
 	<key>BackupEnabled</key>
 	<string>${BACKUP_ENABLED}</string>
+	<key>ForceEncryptionAtRest</key>
+	<string>$(FORCE_ENCRYPTION_AT_REST)</string>
```
Comment (marcomaddy1982) [inline Wire-iOS/Wire-Info.plist:210]: Maybe FORCE_ENCRYPTION_AT_REST_ENABLED

File: Wire-iOS/Sources/Helpers/Bundle+SecurityFlags.swift [hunk 3]
```diff
@@ -55,6 +59,8 @@ enum SecurityFlags {
             return "CameraRollEnabled"
         case .backup:
             return "BackupEnabled"
+        case .forceEncryptionAtRest:
+            return "ForceEncryptionAtRest"
```
Comment (marcomaddy1982) [inline Wire-iOS/Sources/Helpers/Bundle+SecurityFlags.swift:63]: Maybe "ForceEncryptionAtRestEnabled"

**Review by marcomaddy1982 — APPROVED** [review]

File: Wire-iOS/Sources/Helpers/Bundle+SecurityFlags.swift (hunk 3 above)
Comment (johnxnguyen) [reply Wire-iOS/Sources/Helpers/Bundle+SecurityFlags.swift]: Done

File: Wire-iOS/Wire-Info.plist (hunk 2 above)
Comment (johnxnguyen) [reply Wire-iOS/Wire-Info.plist]: Done

File: Wire-iOS/Sources/AppRootViewController.swift (hunk 1 above)
Comment (typfel) [reply Wire-iOS/Sources/AppRootViewController.swift:562]: Yes, I agree it's probably better to handle this in `SessionManager`, but let's refactor it when we add the migration.

**Review by typfel — APPROVED** [review]

**Pull request merged**